x/txfees/keeper/hooks.go

```diff
@@ -14,36 +14,99 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochN
 		_ = osmoutils.ApplyFuncIfNoError(ctx, func(cacheCtx sdk.Context) error {
 			// We allow full slippage. Theres not really an effective way to bound slippage until TWAP's land,
 			// but even then the point is a bit moot.
 			// The only thing that could be done is a costly griefing attack to reduce the amount of osmo given as tx fees.
 			// However the idea of the txfees FeeToken gating is that the pool is sufficiently liquid for that base token.
 			minAmountOut := sdk.ZeroInt()
-			_, err := k.poolManager.SwapExactAmountIn(cacheCtx, nonNativeFeeAddr, feetoken.PoolID, coinBalance, baseDenom, minAmountOut)
+
+			// We swap without charging a taker fee / sending to the non native fee collector, since these are funds that
+			// are accruing from the taker fee itself.
+			_, err := k.poolManager.SwapExactAmountInNoTakerFee(cacheCtx, nonNativeFeeCollectorForStakingRewardsAddr, poolId, coinBalance, baseDenom, minAmountOut)
 			return err
 		})
 	}
 
-	// Get all of the txfee payout denom in the module account
-	baseDenomCoins := sdk.NewCoins(k.bankKeeper.GetBalance(ctx, nonNativeFeeAddr, baseDenom))
+	// Now that the rewards have been swapped, transfer any base denom existing in the non-native fee collector to the fee collector (indirectly distributing to stakers)
+	baseDenomCoins := sdk.NewCoins(k.bankKeeper.GetBalance(ctx, nonNativeFeeCollectorForStakingRewardsAddr, baseDenom))
+	_ = osmoutils.ApplyFuncIfNoError(ctx, func(cacheCtx sdk.Context) error {
+		err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, txfeestypes.NonNativeFeeCollectorForStakingRewardsName, txfeestypes.FeeCollectorName, baseDenomCoins)
+		return err
+	})
+
+	// Non-native fee collector for community pool get swapped entirely into denom specified in the pool manager params.
+
+	poolManagerParams := k.poolManager.GetParams(ctx)
+	denomToSwapTo := poolManagerParams.TakerFeeParams.CommunityPoolDenomToSwapNonWhitelistedAssetsTo
+
+	nonNativeFeeCollectorForCommunityPoolAddr := k.accountKeeper.GetModuleAddress(txfeestypes.NonNativeFeeCollectorForCommunityPoolName)
+	nonNativeFeeCollectorForCommunityPoolBalance := k.bankKeeper.GetAllBalances(ctx, nonNativeFeeCollectorForCommunityPoolAddr)
+
+	// Only non whitelisted assets should exist here since we do direct community pool funds when calculating the taker fee if
+	// the input is a whitelisted asset.
+	for _, coin := range nonNativeFeeCollectorForCommunityPoolBalance {
+		if coin.Denom == denomToSwapTo {
+			continue
+		}
+
+		coinBalance := k.bankKeeper.GetBalance(ctx, nonNativeFeeCollectorForCommunityPoolAddr, coin.Denom)
+		if coinBalance.Amount.IsZero() {
+			continue
+		}
+
+		poolId, err := k.protorevKeeper.GetPoolForDenomPair(ctx, denomToSwapTo, coin.Denom)
+		if err != nil {
+			// The pool route either doesn't exist or is disabled in protorev.
+			// It will just accrue in the non-native fee collector account.
+			// Skip this denom and move on to the next one.
+			continue
+		}
```

Comment:
As it currently stands, we dont want to do this. This is because, in the instance of CL launch, being able to disable osmo was actually really important for the swaps to go through. We have started discussions with skip though to make it so that disabling a route still returns the route when querying, but just shows the route as disabled.

Until this change is in, I do not think its safe to add this.